scripts/send_erc20_packet.js

```diff
@@ -0,0 +1,48 @@
+// This is for emitting SendPacket event directly from Axon to initial the IBC communication
+
+async function main() {
+  // First provider account.
+  const [sender] = await web3.eth.getAccounts();
+  // Receiver should be CKB address, which is first 20 bytes of hash of sender's lock_script
+  const receiver = process.env.RECEIVER;
+  console.log("Sender and receiver:", sender, receiver);
+
+  const ICS20TransferERC20 = await artifacts.require("ICS20TransferERC20Allowlist");
+
+  const transfer = await ICS20TransferERC20.at(
+    process.env.TRANSFER_CONTRACT_ADDRESS
+  );
+
+  const port = "transfer";
+  const channel = process.env.CHANNEL;
+
+  const denom = `${port}/${channel}/${process.env.DENOM}`;
+
+  // Check token associated with the denom that is created before, if not exist, create one
```

Comment:
It's strange that a send transfer script will deploy and mint token. The token should have been transferred from counterparty chain/module.

Reply:
this script serves the Axon -> CKB direction, which means users will use this to send ERC20 token, and this script would create one if users didn't deploy their ERC20 contract before running this script

Reply:
So this is for when axon is the "source zone"?

In this case the denom should be `hex(token contract address)`. The transfer module will use `_transferFrom` instead of `_mint`/`_burn`. It doesn't have anything to do with the `denomTokenContract` mapping, which is for when axon is the “sink zone”.

Note that this isn't supported on the ckb side yet.

Reply:
I mean the case of axon is the "source zone" is when axon received token transfer from ckb, which is ckb has ability to operate the token transferred back from axon